### ISSUE TITLE
Issue when connecting to server

### ISSUE THREAD

**Issue by StevenZiu** [hwdsl2/setup-ipsec-vpn#728]
Keep coming across this error from windows 10:

```The L2TP connection attempt failed because the security layer encountered a processing error during initial negotiations with the remote computer```

I tried all the ways I found from google, like restarting IPsec services, IKE services, and enable CHAP V2 protocols, however, the issue still exists.

When I try to connect from iPhone, it failed as well.

**Comment by Builimacen** [comment]
In Windows 10 you have to use a fix.

I was able to connect ONLY after using this in a reg file:

Windows Registry Editor Version 5.00

[HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\PolicyAgent]
"AssumeUDPEncapsulationContextOnSendRule"=dword:00000002


After that it just connected and had no problems with the VPN.

**Comment by hwdsl2** [comment]
@StevenZiu Please try the troubleshooting steps in [1]. Otherwise, it may indicate a network issue between your VPN client and server.

[1] https://github.com/hwdsl2/setup-ipsec-vpn/blob/master/docs/clients.md#troubleshooting

**Comment by basedtho** [comment]
have you managed to solve this?

**Comment by JagarYousef** [comment]
same here and I have no idea why you think it is included in the troubleshooting and then close it!

**Comment by hwdsl2** [comment]
@JagarYousef Instead of replying here, feel free to open a new issue with the requested details (redacted logs, etc.). The troubleshooting section does include answers to common VPN configuration issues.
